ext/gd/gd.c

```diff
@@ -1912,7 +1944,63 @@ PHP_FUNCTION(imagejpeg)
 /* {{{ Output WBMP image to browser or file */
 PHP_FUNCTION(imagewbmp)
 {
-	_php_image_output_ctx(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_GDIMG_TYPE_WBM, "WBMP", gdImageWBMPCtx);
+	zval *imgind;
+	zend_long foreground_color = -1;
+	gdImagePtr im;
+	int argc = ZEND_NUM_ARGS();
+	int i;
+	gdIOCtx *ctx = NULL;
+	zval *to_zval = NULL;
+	php_stream *stream;
+	int close_stream = 1;
+
+	if (zend_parse_parameters(argc, "O|z!l", &imgind, gd_image_ce, &to_zval, &foreground_color) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	im = php_gd_libgdimageptr_from_zval_p(imgind);
+
+	if (argc > 1 && to_zval != NULL) {
+		if (Z_TYPE_P(to_zval) == IS_RESOURCE) {
```

Comment:
It would be nice to share this stream code in a common function as well. I'm also wondering why imagexbm handles this part differently than the other functions...

Reply:
I think that just nobody implement accepting a stream.

Reply:
I think I managed to extract it. let me know if there's a more elegant way to do so.